salt/modules/state.py

```diff
@@ -783,6 +784,35 @@ def apply_(mods=None, **kwargs):
     return highstate(**kwargs)
 
 
+def test(*args, **kwargs):
+    '''
+    Alias for `state.apply` with the kwarg `test` forced to `True`.
+
+    This is a nicety to avoid the need to type out `test=True` and the possibility of
+    a typo causing changes you do not intend.
+    '''
+    try:
+        test_state_before = __opts__["test"]
+    except KeyError:
+        reset_opts_test = False  # track if __opts__["test"] should be set after run
+    else:
+        reset_opts_test = True
+
+    __opts__["test"] = True
+
+    kwargs["test"] = True
+
+    try:
+        ret = apply_(*args, **kwargs)
+    finally:
+        if reset_opts_test:
```

Comment:
we should refrain from editing `__opts__` at all. You can pass `test` as a kwarg to `apply_` to keep things simpler.

Reply:
Ok cool, I wan't sure about how this was supposed to work. Can you give me insight in to what opts is ever meant to contain then? My guess is just cli (or runner, whatever ran it) arguments? Thanks

Reply:
should be fixed now